pkg/ruler/ruler.go

```diff
@@ -149,59 +139,64 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.DurationVar(&cfg.ResendDelay, "ruler.resend-delay", time.Minute, `Minimum amount of time to wait before resending an alert to Alertmanager.`)
 }
 
+// MultiTenantManager is the interface of interaction with a Manager that is tenant aware.
+type MultiTenantManager interface {
+	// SyncRuleGroups is used to sync the Manager with rules from the RuleStore.
+	SyncRuleGroups(ctx context.Context, ruleGroups map[string]store.RuleGroupList)
+	// GetRules fetches rules for a particular tenant (userID).
+	GetRules(userID string) []*promRules.Group
+	// Stop stops all Manager components.
+	Stop()
```

Comment:
Seeing `Stop` method in the generic interface looks like an anti-pattern to me. As Ruler is not creating the manager, I don't think it should be one stopping it either.

(But that can be addressed in another PR)

Reply:
I agree, we need the `Stop()` function but I had a feeling the other function should be named `StartOrSyncRuleGroups()`. That sounds a little long, but do you have any other suggestions?